src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -624,10 +625,34 @@ public void setNoDamageTicks(int ticks)
 	}
 
 	@Override
-	public Player getKiller()
-	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+	public void kill() {
+		this.health = 0;
+
+		if (!this.alive) return;
+
+		List<ItemStack> drops = Arrays.asList(this.getInventory().getContents());
+		PlayerDeathEvent event = new PlayerDeathEvent(this, drops, 0, getName() + " got killed");
+		Boolean keepInventory = getWorld().getGameRuleValue(GameRule.KEEP_INVENTORY);
```

Comment:
No need to box this boolean
```suggestion
		boolean keepInventory = getWorld().getGameRuleValue(GameRule.KEEP_INVENTORY);
```

Reply:
The reason I used a Boolean is to avoid null pointer exceptions. It should not happen, but I thought it would be better to prevent it

Reply:
Btw, the boolean is boxed by default as it is a generic getter, the unboxing may cause npe if null

Reply:
Ah, the fact that its return value can sometimes be `null` is probably a bug within MockBukkit.

Reply:
So what do we do here? We keep Boolean as it is a best practice to perform null checks?

Reply:
I am working on a rewrite of the gamerules in MockBukkit anyway. So keep it as is for now and let me worry about it later? :eyes:

Reply:
Seems good to me